iambic/config/dynamic_config.py

```diff
@@ -160,6 +161,9 @@ def configured_plugins(self):
 
     async def set_config_secrets(self):
         if self.extends:
+            if len(self.extends) > 1:
+                raise MultipleSecretsNotAcceptedException()
+
             for extend in self.extends:
```

Comment:
Would recommend to corroborate the limitation of 1 extends item in the list that we do away with the for loop and simplify the entry (ie. entry = self.extends[0]...)